adapters/openweb/openweb.go

```diff
@@ -0,0 +1,179 @@
+package openweb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type openwebImpExt struct {
+	OpenWeb openrtb_ext.ExtImpOpenWeb `json:"openweb"`
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	totalImps := len(request.Imp)
+	errors := make([]error, 0, totalImps)
+	sourceIdToImpIds := make(map[int][]int)
+
+	for i := 0; i < totalImps; i++ {
+
+		sourceId, err := validateImpression(&request.Imp[i])
+
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if _, ok := sourceIdToImpIds[sourceId]; !ok {
+			sourceIdToImpIds[sourceId] = make([]int, 0, totalImps-i)
+		}
+
+		sourceIdToImpIds[sourceId] = append(sourceIdToImpIds[sourceId], i)
+
+	}
+
+	totalReqs := len(sourceIdToImpIds)
+	if 0 == totalReqs {
```

Comment:
**Nitpick:** Please consider writing this as `if totalReqs == 0` for ease of readability